### ISSUE TITLE
FILE.dirname returns incorrect results depending on trailing slash

### ISSUE THREAD

**Issue by tolmasky** [tlrobinson/narwhal#46]
irb:

File.dirname("/Users/tolmasky") => "/Users"
File.driname("/Users/tolmasky/") => "/Users"

narwhal:

File.dirname("/Users/tolmasky") => "/Users"
File.driname("/Users/tolmasky/") => "/Users/tolmasky"

**Comment by tlrobinson** [comment]
This one definitely seems like a bug. Assuming for a moment we want to retain the "trailing slash denotes a directory" behavior (I'm not sure if we do) shouldn't dirname return the parent directory of the path passed in, regardless of whether or not it's a directory? (and it should end with a trailing slash since we know the parent is definitely a directory)

**Comment by kriskowal** [comment]
This is definitely a bug.  We probably should take the tolerant of input, strict of output approach.  Let's add the trailing slash for all dirname results, but ignore the terminal slash for the input.  Good idea.

**Comment by tlrobinson** [comment]
I agree I don't think we should continue using the trailing slash to make assumptions about whether it's a directory or not.

Related: tlrobinson/narwhal#47

**Comment by nevilleburnell** [comment]
Added initial tests for dirname()

http://github.com/cloudwork/narwhal/commit/5cc3fc003a8b8e1bc7d8d6b4aa760ff7f04ce9c6

**Comment by tlrobinson** [comment]
tests/commonjs/file/dirname.js has some dirname tests

**Comment by nevilleburnell** [comment]
And one more test case

http://github.com/cloudwork/narwhal/commit/ad9f667723955e27eb49eea4f28df0c50e8e9ad9

**Comment by nevilleburnell** [comment]
oops <grin>

thanks

**Comment by nevilleburnell** [comment]
moved test cases to tests/commonjs/file/dirname.js

http://github.com/cloudwork/narwhal/commit/ead86151ef55fa755547e0cc6afe71c0e45c9c51
